tests/task.test.ts

```diff
@@ -517,11 +517,171 @@ describe.each([{ permission: 'Master' }, { permission: 'Admin' }])(
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toContain('beforeEnqueuedAt')
+    })
+
+    // delete: afterEnqueuedAt
+    test(`${permission} key: Delete a task using afterEnqueuedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
+      const enqueuedTask = await client.deleteTasks({
+        afterEnqueuedAt: currentTime,
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toContain('afterEnqueuedAt')
+    })
+
+    // delete: beforeStartedAt
+    test(`${permission} key: Delete a task using beforeStartedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
+      const enqueuedTask = await client.deleteTasks({
+        beforeStartedAt: currentTime,
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toContain('beforeStartedAt')
+    })
+
+    // delete: afterStartedAt
+    test(`${permission} key: Delete a task using afterStartedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
+      const enqueuedTask = await client.deleteTasks({
+        afterStartedAt: currentTime,
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toContain('afterStartedAt')
+    })
+
+    // delete: beforeFinishedAt
+    test(`${permission} key: Delete a task using beforeFinishedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
+      const enqueuedTask = await client.deleteTasks({
+        beforeFinishedAt: currentTime,
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
+      expect(task.details?.originalQuery).toContain('beforeFinishedAt')
+    })
+
+    // delete: afterFinishedAt
+    test(`${permission} key: Delete a task using afterFinishedAt filter`, async () => {
+      const client = await getClient(permission)
+
+      const currentTimeStamp = Date.now()
+      const currentTime = new Date(currentTimeStamp)
+      const enqueuedTask = await client.deleteTasks({
+        afterFinishedAt: currentTime,
+      })
+      const task = await client.waitForTask(enqueuedTask.taskUid)
+
+      expect(task.type).toEqual(TaskTypes.TASK_DELETION)
```

Comment:
Did you miss this one?
`expect(task.details?.originalQuery).toContain('afterFinishedAt')`